### PR TITLE
feat: impl `EmptyMetric` plan and `time()` function

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

`EmptyMetric` is like `EmptyRelation`, it generates record batch by itself and doesn't need input.

[`time()`](https://prometheus.io/docs/prometheus/latest/querying/functions/#time) function is from PromQL. It return the timestamp as value.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/promql/src/extension_plan/empty_metric.rs [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+            ),
+        }
+    }
+
+    fn metrics(&self) -> Option<MetricsSet> {
+        Some(self.metric.clone_inner())
+    }
+
+    fn statistics(&self) -> Statistics {
+        let estimated_row_num = (self.end - self.start) as f64 / self.interval as f64;
```
Comment (yuanbohan) [inline src/promql/src/extension_plan/empty_metric.rs:196]: Is it necessary to limit the maximum row number?

Comment (waynexia) [reply src/promql/src/extension_plan/empty_metric.rs:196]: This is only for statistics. 

The problem that too many rows are calculated inside one batch should be fixed in someplace else, but I haven't figured out how to do it properly 🤧

File: src/promql/src/planner.rs [hunk 2]
```diff
@@ -331,6 +337,26 @@ impl PromPlanner {
                 })
             }
             PromExpr::Call(Call { func, args }) => {
+                // TODO(ruihang): refactor this, transform the AST in advance to include a empty metric table.
```
Comment (killme2008) [inline src/promql/src/planner.rs:340]: typo: an empty

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by yuanbohan — APPROVED** [review]

File: src/promql/src/planner.rs (hunk 2 above)
Comment (waynexia) [reply src/promql/src/planner.rs]: 👍

**Pull request merged**